core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -564,12 +567,20 @@ private Query showCreateMaterializedView(ShowCreate node)
                     false,
                     viewDefinition.get().getGracePeriod()
                             .map(DateTimeUtils::formatDayTimeInterval),
-                    Optional.empty(), // TODO support WHEN STALE
+                    Optional.of(toSqlWhenStaleBehavior(viewDefinition.get().getWhenStaleBehavior().orElse(INLINE))),
```

Comment:
we should change the CreateMaterializedView field (in the non-connector class) to be non-Optional.

we could also change that in ConnectorMaterializedViewDefinition. in fact, backwards compatibility was broken, as the ConnectorMaterializedViewDefinition constructor already requires passing in `Optional<WhenStaleBehavior>`, so it could as well take `WhenStaleBehavior`